tests/server/test_server.py

```diff
@@ -363,17 +354,76 @@ def test_send_500_on_revoke_failure(
                 # logout halted due to token revoke fail
                 response = self.client.post(
                     self.url_path,
-                    headers={
-                        "x-access-token": token,
-                    },
+                    headers={"x-access-token": token},
                 )
                 assert response.status_code == 500
 
             def test_send_201_on_success(self, token: str) -> None:
                 response = self.client.post(
                     self.url_path,
-                    headers={
-                        "x-access-token": token,
-                    },
+                    headers={"x-access-token": token},
                 )
                 assert response.status_code == 201
+
+    class Test_Wallet:
+        class Test_DepositEndpoint:
+            @pytest.fixture(autouse=True)
+            def prepare_tests(self, client: FlaskClient) -> None:
+                self.register_path: str = "api/v1/auth/register"
+                self.login_path: str = "api/v1/auth/login"
+                self.url_path: str = "api/v1/wallet/deposit"
+                self.client: FlaskClient = client
+
+            @pytest.fixture(name="token")
+            def fixture_register_and_login(self) -> str:
+                self.client.post(
+                    self.register_path,
+                    json={"email": "legit_email@gmail.com", "password": "thelegend27"},
+                )
+                login_response = self.client.post(
+                    self.login_path,
+                    json={"email": "legit_email@gmail.com", "password": "thelegend27"},
+                )
+                return login_response.get_json()["auth_token"]
+
+            def test_send_200_on_success(self, token: str) -> None:
+                response = self.client.post(
+                    self.url_path,
+                    json={"amount": 5.75},
+                    headers={"x-access-token": token},
+                )
+                assert response.status_code == 200
+
+            def test_send_400_on_invalid_json_format(self, token: str) -> None:
+                response = self.client.post(
+                    self.url_path,
+                    data={"amount": -5.75},
+                    headers={"x-access-token": token},
+                )
+                assert response.status_code == 400
```

Comment:
Since JSONSchema uses `exclusiveMinimum: 0`, can you add the case where `0` amount is passed?

```suggestion
            @pytest.mark.parametrize("amount", [(-5.75), (0.0)])
            def test_send_400_on_invalid_json_format(self, token: str, amount: float) -> None:
                response = self.client.post(
                    self.url_path,
                    data={"amount": amount},
                    headers={"x-access-token": token},
                )
                assert response.status_code == 400
```

Check whether this code works, I've typed it in off the top of my head.